controllers/articles.controller.js

```diff
@@ -12,13 +12,30 @@ exports.getArticles = (req, res, next) => {
 
 
 exports.getArticleById = (req, res, next) => {
-    const {article_id} = req.params
-    selectArticleById(article_id)
+    selectArticleById(req.params)
     .then((article) => {
-        res.status(200).send({article})
+        if(article.length === 0){
+            return Promise.reject({status: 404, msg : 'Not Found'})
+        } 
+        res.status(200).send({article: article[0]})
     })
     .catch((err) => {
         next(err)
     })
 }
 
+exports.patchArticleById = (req, res, next) => {
+
+    updateArticle(req.params, req.body)
+    .then((article) => {
+        if(article[0] === undefined){
```

Comment:
Ideally this error handling would be in your model.
 Handling errors related to data storage, retrieval, and validation in the model layer helps maintain a clean separation of concerns.